app/views/reports/regions/show.html.erb

```diff
@@ -364,7 +364,11 @@
                   <p class="m-0px c-black c-print-black">
                     total <%= "registration".pluralize(@data.last_value(:cumulative_registrations)) %><br>till
                     <span data-registrations-period-end="<%= @period.bp_control_range_end_date %>">
-                      <%= @period.bp_control_range_end_date %>
+                      <% if @period.value.strftime("%m") == Date.today.strftime("%m") %>
+                        <%= Date.today.to_s(:day_mon_year) %>
+                      <% else %>
+                        <%= @period.bp_control_range_end_date %>
+                      <% end %>
```

Comment:
I think this can be simplified a bit ...its probably worth moving to a helper method as well:

```ruby
module DashboardHelper
  def registrations_end_date
    if @period.value.month == Date.current.month
      <%= Date.current.to_s(:day_mon_year)
    else
      @period.bp_control_range_end_date
    end
  end
```

Note that you probably want to use `Date.current`, and not `Date.today`, because of [timezones](https://thoughtbot.com/blog/its-about-time-zones).

Reply:
Thanks @rsanheim!